homeworks/olya.kosenko_olyabot/homework_6/Task_6_2.js

```diff
@@ -0,0 +1,77 @@
+class Horse {
+    constructor(name, breed) {
+        this.name = name;
+        this.breed = breed;
+    }
+}
+class Racer extends Horse {
+    constructor(name, breed) {
+        super(name, breed);
+        this.distance = 0;
+        this.speed = 0;
+    }
+
+    run() {
+        const t1 = setInterval(() => {
+            this.distance += this.speed;
+            this.setSpeed();
+        }, 1000);
+        setTimeout(() => {
+            clearInterval(t1);
+        }, 10000);
```

Comment:
I would redesign this... horse just needs to run, horse doesn't know when to stop, hippodrom decides when to stop